-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-6020d2/21000/summary.html
**COMMIT**: 9a74c283df87ba615c2d6b786925953743185e7b
**CMSSW**: CMSSW_12_2_X_2021-12-05-0000/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36366/21000/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>Copying tmp/slc7_amd64_gcc900/src/RecoEgamma/EgammaTools/src/RecoEgammaEgammaTools/libRecoEgammaEgammaTools.so to productstore area:
Leaving library rule at RecoEgamma/EgammaTools
Entering library rule at src/RecoEgamma/EgammaTools/test
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/src/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/src/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc: In member function 'virtual void GBRWrapperMaker::analyze(const edm::Event&, const edm::EventSetup&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/src/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc:67:16: error: expected primary-expression before '(' token
   67 |   edm::LogPrint(GBRWrapperMaker) << "Load forest";
      |                ^
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/src/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc:67:32: error: expected primary-expression before ')' token
   67 |   edm::LogPrint(GBRWrapperMaker) << "Load forest";
      |                                ^
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Creating project symlinks
>> Entering Package RecoEgamma/EgammaTools
>> Compile sequence completed for CMSSW CMSSW_12_2_X_2021-12-05-0000
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --ignoreWarning=Wdeprecated-declarations --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/tmp/slc7_amd64_gcc900/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package RecoEgamma/EgammaTools
Entering library rule at src/RecoEgamma/EgammaTools/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_2_X_2021-12-05-0000/src/RecoEgamma/EgammaTools/plugins/CalibratedElectronProducers.cc
</pre>

--- a/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc
+++ b/RecoEgamma/EgammaTools/test/GBRWrapperMaker.cc
@@ -1,4 +1,3 @@
-
 // -*- C++ -*-
 //
 // Package:    GBRWrapperMaker
@@ -22,7 +21,7 @@
 
 // user include files
 #include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
@@ -41,10 +40,10 @@
 // class declaration
 //
 
-class GBRWrapperMaker : public edm::EDAnalyzer {
+class GBRWrapperMaker : public edm::one::EDAnalyzer<> {
 public:
   explicit GBRWrapperMaker(const edm::ParameterSet&);
-  ~GBRWrapperMaker();
+  ~GBRWrapperMaker() = default;
 
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
@@ -52,36 +51,9 @@ private:
   virtual void beginJob() override;
   virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
   virtual void endJob() override;
-
-  virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
-  virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
-  virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
-  virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
-
-  // ----------member data ---------------------------
 };
 
-//
-// constants, enums and typedefs
-//
-
-//
-// static data member definitions
-//
-
-//
-// constructors and destructor
-//
-GBRWrapperMaker::GBRWrapperMaker(const edm::ParameterSet& iConfig)
-
-{
-  //now do what ever initialization is needed
-}
-
-GBRWrapperMaker::~GBRWrapperMaker() {
-  // do anything here that needs to be done at desctruction time
-  // (e.g. close files, deallocate resources etc.)
-}
+GBRWrapperMaker::GBRWrapperMaker(const edm::ParameterSet& iConfig) {}
 
 //
 // member functions
@@ -91,26 +63,20 @@ GBRWrapperMaker::~GBRWrapperMaker() {
 void GBRWrapperMaker::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   using namespace edm;
 
-  //from Josh:
-  //TFile *infile = new TFile("../data/GBRLikelihood_Clustering_746_bx25_Electrons_NoPosition_standardShapes_NoPS_PFMustache_results_PROD.root","READ");
   TFile* infile = new TFile("../data/GBRLikelihood_Clustering_746_bx25_HLT.root", "READ");
-  printf("load forest\n");
-
-  //GBRForestD *p4comb = (GBRForest*)infile->Get("CombinationWeight");
+  edm::LogPrint("GBRWrapperMaker") << "Load forest";
   GBRForestD* gbreb = (GBRForestD*)infile->Get("EBCorrection");
   GBRForestD* gbrebvar = (GBRForestD*)infile->Get("EBUncertainty");
   GBRForestD* gbree = (GBRForestD*)infile->Get("EECorrection");
   GBRForestD* gbreevar = (GBRForestD*)infile->Get("EEUncertainty");
 
-  printf("made objects\n");
+  edm::LogPrint("GBRWrapperMaker") << "Make objects";
   edm::Service<cond::service::PoolDBOutputService> poolDbService;
   if (poolDbService.isAvailable()) {
-    //poolDbService->writeOne( p4comb, poolDbService->beginOfTime(),
-    //		     "gedelectron_p4combination"  );
-    poolDbService->writeOne(gbreb, poolDbService->beginOfTime(), "mustacheSC_online_EBCorrection");
-    poolDbService->writeOne(gbrebvar, poolDbService->beginOfTime(), "mustacheSC_online_EBUncertainty");
-    poolDbService->writeOne(gbree, poolDbService->beginOfTime(), "mustacheSC_online_EECorrection");
-    poolDbService->writeOne(gbreevar, poolDbService->beginOfTime(), "mustacheSC_online_EEUncertainty");
+    poolDbService->writeOneIOV(*gbreb, poolDbService->beginOfTime(), "mustacheSC_online_EBCorrection");
+    poolDbService->writeOneIOV(*gbrebvar, poolDbService->beginOfTime(), "mustacheSC_online_EBUncertainty");
+    poolDbService->writeOneIOV(*gbree, poolDbService->beginOfTime(), "mustacheSC_online_EECorrection");
+    poolDbService->writeOneIOV(*gbreevar, poolDbService->beginOfTime(), "mustacheSC_online_EEUncertainty");
   }
 }
 
@@ -119,18 +85,6 @@ void GBRWrapperMaker::beginJob() {}
 
 // ------------ method called once each job just after ending the event loop  ------------
 void GBRWrapperMaker::endJob() {}
-
-// ------------ method called when starting to processes a run  ------------
-void GBRWrapperMaker::beginRun(edm::Run const&, edm::EventSetup const&) {}
-
-// ------------ method called when ending the processing of a run  ------------
-void GBRWrapperMaker::endRun(edm::Run const&, edm::EventSetup const&) {}
-
-// ------------ method called when starting to processes a luminosity block  ------------
-void GBRWrapperMaker::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
-
-// ------------ method called when ending the processing of a luminosity block  ------------
-void GBRWrapperMaker::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {}
 
 // ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
 void GBRWrapperMaker::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {

